components/xperiencify/app/xperiencify.app.ts

```diff
@@ -1,13 +1,170 @@
 import { defineApp } from "@pipedream/types";
+import { axios } from "@pipedream/platform";
 
 export default defineApp({
   type: "app",
   app: "xperiencify",
-  propDefinitions: {},
+  propDefinitions: {
+    courseId: {
+      type: "integer",
+      label: "Course",
+      description: "Course",
+      async options() {
+        const courses = await this.listCourses();
+        return courses.map((course) => ({
+          label: course.title,
+          value: course.id,
+        }));
+      },
+    },
+    student: {
+      type: "string",
+      label: "Student",
+      description: "The student's email address",
+      async options({ courseId }) {
+        const students = !courseId
+          ? await this.listAllStudents()
+          : await this.getStudentsForCourse({
+            courseId,
+          });
+        return students.map((student) => ({
+          label: student.first_name,
+          value: student.email,
+        }));
+      },
+    },
+    tags: {
+      type: "string[]",
+      label: "Tags",
+      description: "Student tags",
+      async options({ studentEmail }) {
+        const tags = await this.listStudentTags({
+          studentEmail,
+        });
+        return tags;
+      },
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async _makeRequest({
+      $ = this, method = "get", path, params, data, ...opts
+    }) {
+      try {
+        return await axios($, {
+          url: `https://api.xperiencify.io/api/public${path}`,
+          method,
+          params: {
+            api_key: this.$auth.api_key,
```

Comment:
Move this to a getter method.

```suggestion
            api_key: this._apiKey(), // this.$auth.api_key
```